lib/src/view/analysis/retro_screen.dart

```diff
@@ -0,0 +1,391 @@
+import 'package:auto_size_text/auto_size_text.dart';
+import 'package:chessground/chessground.dart';
+import 'package:dartchess/dartchess.dart';
+import 'package:fast_immutable_collections/fast_immutable_collections.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:lichess_mobile/src/model/analysis/analysis_preferences.dart';
+import 'package:lichess_mobile/src/model/analysis/retro_controller.dart';
+import 'package:lichess_mobile/src/model/common/node.dart';
+import 'package:lichess_mobile/src/model/engine/evaluation_preferences.dart';
+import 'package:lichess_mobile/src/model/settings/board_preferences.dart';
+import 'package:lichess_mobile/src/model/settings/general_preferences.dart';
+import 'package:lichess_mobile/src/styles/styles.dart';
+import 'package:lichess_mobile/src/utils/l10n_context.dart';
+import 'package:lichess_mobile/src/utils/navigation.dart';
+import 'package:lichess_mobile/src/view/analysis/analysis_board.dart';
+import 'package:lichess_mobile/src/view/analysis/analysis_layout.dart';
+import 'package:lichess_mobile/src/view/analysis/retro_settings_screen.dart';
+import 'package:lichess_mobile/src/view/engine/engine_depth.dart';
+import 'package:lichess_mobile/src/view/engine/engine_gauge.dart';
+import 'package:lichess_mobile/src/view/engine/engine_lines.dart';
+import 'package:lichess_mobile/src/view/puzzle/puzzle_feedback_widget.dart';
+import 'package:lichess_mobile/src/view/settings/toggle_sound_button.dart';
+import 'package:lichess_mobile/src/widgets/bottom_bar.dart';
+import 'package:lichess_mobile/src/widgets/buttons.dart';
+import 'package:lichess_mobile/src/widgets/pgn.dart';
+import 'package:lichess_mobile/src/widgets/platform_context_menu_button.dart';
+
+class RetroScreen extends ConsumerWidget {
+  const RetroScreen({required this.options, super.key});
+
+  final RetroOptions options;
+
+  static Route<dynamic> buildRoute(BuildContext context, RetroOptions options) {
+    return buildScreenRoute(context, screen: RetroScreen(options: options));
+  }
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final enginePrefs = ref.watch(engineEvaluationPreferencesProvider);
+    final asyncState = ref.watch(retroControllerProvider(options));
+
+    final appBarActions = asyncState.hasValue
+        ? [
+            if (asyncState.requireValue.isEngineAvailable(enginePrefs) == true)
+              EngineDepth(
+                savedEval: asyncState.valueOrNull?.currentNode.eval,
+                goDeeper: () =>
+                    ref.read(retroControllerProvider(options).notifier).requestEval(goDeeper: true),
+              ),
+            _RetroMenu(options: options),
+          ]
+        : <Widget>[];
+
+    return Scaffold(
+      appBar: AppBar(title: Text(context.l10n.learnFromYourMistakes), actions: appBarActions),
```

Comment:
You can use `AppBarTitleText` to auto size the title so it doesn't appear truncated.

Reply:
Maybe we can even append the error index after the title, like `Learn from your mistakes 2/4`.